### ISSUE TITLE
Attribute configuration no longer allows multiple attributes per pattern

### ISSUE THREAD

**Issue by cab** [tokio-rs/prost#490]
The `type_attributes`, `field_attributes`, etc methods are described as cumulative, but after this change there can only be one attribute per path.

It's not clear to me if this new behavior is the correct behavior or if the old behavior was unintentionally allowing this.

I _think_ this is a regression from https://github.com/tokio-rs/prost/commit/79f0dfd8ed703fcce27f053c3b29c360c7ead5cb. Previously, [multiple values could exist](https://github.com/tokio-rs/prost/commit/79f0dfd8ed703fcce27f053c3b29c360c7ead5cb#diff-702d1c883416266db565804e8243178a5e3b59e9c22dff7d3161247d13aad208L401) with the same matcher, but [now only one entry](https://github.com/tokio-rs/prost/commit/79f0dfd8ed703fcce27f053c3b29c360c7ead5cb#diff-5c6fd8964f3a708b4b4ae0231791a3119fc994c82c498c8a69af388567dbdd41R14) can exist in the `PathMap`'s `matchers` hashmap.

### Example

```rust
.type_attributes(".", "[derive(::serde::Serialize)]")
.type_attributes(".", "[serde(rename = \"camelCase\")]")
```

Before the regression (e.g. on `0.7`), this would output

```rust
#[derive(::serde::Serialize)]
#[serde(rename = "camelCase")]
struct MyProto {}
```

After the regression, this outputs
```rust
#[serde(rename = "camelCase")]
struct MyProto {}
```

**Comment by cab** [comment]
if this is a regression, one possible fix might be making `Config#type_attributes` a `PathMap<Vec<String>>` and then adding something like this to `PathMap`:
```rust
pub(crate) fn extend<A>(&mut self, matcher: String, value: T) where T: Extend<A> + IntoIterator<Item = A> + Default {
    self.matchers.entry(matcher).or_default().extend(value);
}
```

**Comment by LucioFranco** [comment]
Fixed in #522
